lib/logstash/inputs/jdbc.rb

```diff
@@ -163,22 +163,27 @@ def run(queue)
       @scheduler.cron @schedule do
         execute_query(queue)
       end
+
       @scheduler.join
     else
       execute_query(queue)
     end
   end # def run
 
-  def teardown
+  def stop
+    super
     @scheduler.stop if @scheduler
```

Comment:
In a similar situation at https://github.com/logstash-plugins/logstash-input-neo4j/pull/4/files#diff-6ec72c1ec89ef0f4ba329099faaa459fR60 I used the shutdown method, do you know what works better?